xalia/Win32/HwndProvider.cs

```diff
@@ -113,16 +125,95 @@ public Task<(bool, int, int)> GetClickablePointAsync(UiDomElement element)
 
         public string[] GetTrackedProperties()
         {
-            return null;
+            return tracked_properties;
         }
 
         public void NotifyElementRemoved(UiDomElement element)
         {
             Element = null;
         }
 
+        private void WatchChildren()
+        {
+            if (_watchingChildren)
+                return;
+            if (Element.MatchesDebugCondition())
+                Utils.DebugWriteLine($"WatchChildren for {Element} (win32)");
+            _watchingChildren = true;
+            Utils.RunIdle(PollChildren); // need to wait for any other providers to remove their children
+        }
+
+        private void PollChildren()
+        {
+            if (!_watchingChildren)
+                return;
+
+            var child_hwnds = EnumImmediateChildWindows(Hwnd);
+
+            // First remove any existing children that are missing or out of order
+            int i = 0;
+            foreach (var new_child in child_hwnds)
+            {
+                var child_name = $"hwnd-{new_child}";
+                if (!Element.Children.Exists((UiDomElement element) => element.DebugId == child_name))
+                    continue;
+                while (Element.Children[i].DebugId != child_name)
+                {
+                    Element.RemoveChild(i);
+                    _childCount--;
+                }

```

Comment:
You do `Element.RemoveChild` and `Element.AddChild` alongside incrementing/decrementing `_childCount` in a few places here, could there maybe be some kind of helper function for this? Mainly to avoid a circumstance where you accidentally add/remove a child and forget to increment/decrement the `_childCount` variable.